perception_dataset/rosbag2/rosbag2_to_non_annotated_t4_converter.py

```diff
@@ -583,14 +613,16 @@ def _convert_image(
                 misc_utils.nusc_timestamp_to_unix_timestamp(sample_record.timestamp)
                 for sample_record in sample_records
             ]
+            lidar_to_camera_latency_sec = (
```

Comment:
Should we consider omitting the addition of self._camera_lidar_latency_margin for the sync parameter?

Reply:
Understood. But at least empirically we should make `lidar_to_camera_latency_sec` to `0.15` from several Rosbags including the one used in pytest in this repository. Since `self._camera_latency` is something the users are already providing in YAML file, do you think it make sense to change as follows
```
            lidar_to_camera_latency_sec = (
                self._lidar_latency + self._camera_latency
            )
```
and
```
lidar_latency_sec: float = 0.04
```
?

Reply:
(FYI lidar_latency_sec: float = 0.03 in current implementation)

Reply:
@kminoda 
Is it like
```

def get_lidar_camera_synced_frame_info(
    image_timestamp_list: List[float],
    lidar_timestamp_list: List[float],
    lidar_to_camera_latency_sec: float = 0.0,
    camera_jitter_sec: float = 0.0,
    system_scan_period: float = 0.1,
    num_load_frames: int = 0,
    msg_display_interval: int = 100,
):
    synced_frame_info_list: List[int, int, float] = []

    current_image_index: int = 0
    for lidar_index, lidar_timestamp in enumerate(lidar_timestamp_list):
        if lidar_index >= num_load_frames:
            break
        image_timestamp = image_timestamp_list[current_image_index]

        while image_timestamp - lidar_timestamp < lidar_to_camera_latency_sec:
            current_image_index += 1
            image_timestamp = image_timestamp_list[current_image_index]

        if image_timestamp - lidar_timestamp > system_scan_period + lidar_to_camera_latency_sec + camera_jitter_sec:
            # Image is dropped
            dummy_timestamp = image_timestamp - system_scan_period
            synced_frame_info_list.append([None, lidar_index, dummy_timestamp])
            continue

        synced_frame_info_list.append([current_image_index, lidar_index, None])
        current_image_index += 1

        if lidar_index % msg_display_interval == 0:
            print(
                f"frame{lidar_index}, stamp = {image_timestamp}, diff cam - lidar = {image_timestamp - lidar_timestamp:0.3f} sec"
            )
    return synced_frame_info_list
```
where 
`lidar_to_camera_latency_sec = self._lidar_latency + self._camera_latency (=0.6)`

Reply:
I updated the PR. Now the matching algorithm is based on the pre-defined delay specified for each camera in the given config file. Would you check the latest implementation again? :pray: